packages/flutter_tools/templates/list_detail_app/lib/src/app.dart.tmpl

```diff
@@ -0,0 +1,73 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:{{projectName}}/src/dummy_feature/dummy_item_details_view.dart';
+import 'package:{{projectName}}/src/dummy_feature/dummy_item_list_view.dart';
+import 'package:{{projectName}}/src/settings/settings_controller.dart';
+import 'package:{{projectName}}/src/settings/settings_view.dart';
+
+/// The Widget that configures your application.
+class MyApp extends StatelessWidget {
+  const MyApp({
+    Key key,
+    @required this.settingsController,
+  }) : super(key: key);
+
+  final SettingsController settingsController;
+
+  @override
+  Widget build(BuildContext context) {
+    // Glue the SettingsController to the MaterialApp.
+    //
+    // The AnimatedBuilder Widget listens to the SettingsController for changes.
+    // Whenever the user updates their settings, the MaterialApp is rebuilt.
+    return AnimatedBuilder(
+      animation: settingsController,
+      builder: (BuildContext context, Widget child) {
+        return MaterialApp(
+          // Providing a restorationScopeId allows the Navigator built by the
+          // MaterialApp to restore the navigation stack when a user returns to
+          // the app.
+          restorationScopeId: 'app',
+
+          // Provide the generated AppLocalizations to the MaterialApp. This
+          // allows descendant Widgets to display the correct translations
+          // depending on the user's locale.
+          localizationsDelegates: [AppLocalizations.delegate],
+
+          // Use AppLocalizations to configure a the correct application title
+          // depending on the user's locale.
+          //
+          // The appTitle is defined in .arb files found in the localization
+          // directory.
+          onGenerateTitle: (context) => AppLocalizations.of(context).appTitle,
+
+          // Define a light and dark color theme. Then, read the user's
+          // preferred ThemeMode (light, dark or system default) from the
```

Comment:
nit:

```suggestion
          // preferred ThemeMode (light, dark, or system default) from the
```

Reply:
Haha, I thought someone might make an Oxford-comma call out! I don't know if this sentence is unclear without it, but added it nonetheless :)